### ISSUE TITLE
Bug: uninitialized constant #<Class:RuboCop::Cop::AlignmentCorrector>::StringIO

### ISSUE THREAD

**Issue by Spence1115** [rubocop/rubocop#13475]
Bug running rubocop on my gem

--------

## Expected behavior

Rubocop should work

## Actual behavior

Rubocop errors

Debug:
```bash
uninitialized constant #<Class:RuboCop::Cop::AlignmentCorrector>::StringIO
/Users/me/.rbenv/versions/3.3.5/lib/ruby/gems/3.3.0/gems/rubocop-1.66.1/lib/rubocop/cop/correctors/alignment_corrector.rb:101:in `remove'
/Users/me/.rbenv/versions/3.3.5/lib/ruby/gems/3.3.0/gems/rubocop-1.66.1/lib/rubocop/cop/correctors/alignment_corrector.rb:50:in `autocorrect_line'
/Users/me/.rbenv/versions/3.3.5/lib/ruby/gems/3.3.0/gems/rubocop-1.66.1/lib/rubocop/cop/correctors/alignment_corrector.rb:25:in `block in correct'
/Users/me/.rbenv/versions/3.3.5/lib/ruby/gems/3.3.0/gems/rubocop-1.66.1/lib/rubocop/cop/correctors/alignment_corrector.rb:113:in `block in each_line'
/Users/me/.rbenv/versions/3.3.5/lib/ruby/gems/3.3.0/gems/rubocop-1.66.1/lib/rubocop/cop/correctors/alignment_corrector.rb:112:in `each_line'
/Users/me/.rbenv/versions/3.3.5/lib/ruby/gems/3.3.0/gems/rubocop-1.66.1/lib/rubocop/cop/correctors/alignment_corrector.rb:112:in `each_line'
/Users/me/.rbenv/versions/3.3.5/lib/ruby/gems/3.3.0/gems/rubocop-1.66.1/lib/rubocop/cop/correctors/alignment_corrector.rb:24:in `correct'
/Users/me/.rbenv/versions/3.3.5/lib/ruby/gems/3.3.0/gems/rubocop-1.66.1/lib/rubocop/cop/layout/argument_alignment.rb:109:in `autocorrect'
/Users/me/.rbenv/versions/3.3.5/lib/ruby/gems/3.3.0/gems/rubocop-1.66.1/lib/rubocop/cop/mixin/alignment.rb:80:in `block in register_offense'
/Users/me/.rbenv/versions/3.3.5/lib/ruby/gems/3.3.0/gems/rubocop-1.66.1/lib/rubocop/cop/base.rb:416:in `correct'
/Users/me/.rbenv/versions/3.3.5/lib/ruby/gems/3.3.0/gems/rubocop-1.66.1/lib/rubocop/cop/base.rb:210:in `add_offense'
/Users/me/.rbenv/versions/3.3.5/lib/ruby/gems/3.3.0/gems/rubocop-1.66.1/lib/rubocop/cop/mixin/alignment.rb:79:in `register_offense'
/Users/me/.rbenv/versions/3.3.5/lib/ruby/gems/3.3.0/gems/rubocop-1.66.1/lib/rubocop/cop/mixin/alignment.rb:39:in `block in check_alignment'
/Users/me/.rbenv/versions/3.3.5/lib/ruby/gems/3.3.0/gems/rubocop-1.66.1/lib/rubocop/cop/mixin/alignment.rb:51:in `block in each_bad_alignment'
/Users/me/.rbenv/versions/3.3.5/lib/ruby/gems/3.3.0/gems/rubocop-1.66.1/lib/rubocop/cop/mixin/alignment.rb:47:in `each'
/Users/me/.rbenv/versions/3.3.5/lib/ruby/gems/3.3.0/gems/rubocop-1.66.1/lib/rubocop/cop/mixin/alignment.rb:47:in `each_bad_alignment'
/Users/me/.rbenv/versions/3.3.5/lib/ruby/gems/3.3.0/gems/rubocop-1.66.1/lib/rubocop/cop/mixin/alignment.rb:29:in `check_alignment'
/Users/me/.rbenv/versions/3.3.5/lib/ruby/gems/3.3.0/gems/rubocop-1.66.1/lib/rubocop/cop/layout/argument_alignment.rb:61:in `on_send'
/Users/me/.rbenv/versions/3.3.5/lib/ruby/gems/3.3.0/gems/rubocop-1.66.1/lib/rubocop/cop/commissioner.rb:107:in `public_send'
/Users/me/.rbenv/versions/3.3.5/lib/ruby/gems/3.3.0/gems/rubocop-1.66.1/lib/rubocop/cop/commissioner.rb:107:in `block (2 levels) in trigger_responding_cops'
/Users/me/.rbenv/versions/3.3.5/lib/ruby/gems/3.3.0/gems/rubocop-1.66.1/lib/rubocop/cop/commissioner.rb:171:in `with_cop_error_handling'
/Users/me/.rbenv/versions/3.3.5/lib/ruby/gems/3.3.0/gems/rubocop-1.66.1/lib/rubocop/cop/commissioner.rb:106:in `block in trigger_responding_cops'
/Users/me/.rbenv/versions/3.3.5/lib/ruby/gems/3.3.0/gems/rubocop-1.66.1/lib/rubocop/cop/commissioner.rb:105:in `each'
/Users/me/.rbenv/versions/3.3.5/lib/ruby/gems/3.3.0/gems/rubocop-1.66.1/lib/rubocop/cop/commissioner.rb:105:in `trigger_responding_cops'
/Users/me/.rbenv/versions/3.3.5/lib/ruby/gems/3.3.0/gems/rubocop-1.66.1/lib/rubocop/cop/commissioner.rb:69:in `on_send'
/Users/me/.rbenv/versions/3.3.5/lib/ruby/gems/3.3.0/gems/rubocop-ast-1.36.1/lib/rubocop/ast/traversal.rb:154:in `on_def'
/Users/me/.rbenv/versions/3.3.5/lib/ruby/gems/3.3.0/gems/rubocop-1.66.1/lib/rubocop/cop/commissioner.rb:71:in `on_def'
/Users/me/.rbenv/versions/3.3.5/lib/ruby/gems/3.3.0/gems/rubocop-ast-1.36.1/lib/rubocop/ast/traversal.rb:137:in `block in on_dstr'
/Users/me/.rbenv/versions/3.3.5/lib/ruby/gems/3.3.0/gems/rubocop-ast-1.36.1/lib/rubocop/ast/traversal.rb:137:in `each'
/Users/me/.rbenv/versions/3.3.5/lib/ruby/gems/3.3.0/gems/rubocop-ast-1.36.1/lib/rubocop/ast/traversal.rb:137:in `on_dstr'
/Users/me/.rbenv/versions/3.3.5/lib/ruby/gems/3.3.0/gems/rubocop-1.66.1/lib/rubocop/cop/commissioner.rb:71:in `on_begin'
/Users/me/.rbenv/versions/3.3.5/lib/ruby/gems/3.3.0/gems/rubocop-ast-1.36.1/lib/rubocop/ast/traversal.rb:154:in `on_class'
/Users/me/.rbenv/versions/3.3.5/lib/ruby/gems/3.3.0/gems/rubocop-1.66.1/lib/rubocop/cop/commissioner.rb:71:in `on_class'
/Users/me/.rbenv/versions/3.3.5/lib/ruby/gems/3.3.0/gems/rubocop-ast-1.36.1/lib/rubocop/ast/traversal.rb:138:in `on_while'
/Users/me/.rbenv/versions/3.3.5/lib/ruby/gems/3.3.0/gems/rubocop-1.66.1/lib/rubocop/cop/commissioner.rb:71:in `on_module'
/Users/me/.rbenv/versions/3.3.5/lib/ruby/gems/3.3.0/gems/rubocop-ast-1.36.1/lib/rubocop/ast/traversal.rb:138:in `on_while'
/Users/me/.rbenv/versions/3.3.5/lib/ruby/gems/3.3.0/gems/rubocop-1.66.1/lib/rubocop/cop/commissioner.rb:71:in `on_module'
/Users/me/.rbenv/versions/3.3.5/lib/ruby/gems/3.3.0/gems/rubocop-ast-1.36.1/lib/rubocop/ast/traversal.rb:138:in `on_while'
/Users/me/.rbenv/versions/3.3.5/lib/ruby/gems/3.3.0/gems/rubocop-1.66.1/lib/rubocop/cop/commissioner.rb:71:in `on_module'
/Users/me/.rbenv/versions/3.3.5/lib/ruby/gems/3.3.0/gems/rubocop-ast-1.36.1/lib/rubocop/ast/traversal.rb:20:in `walk'
/Users/me/.rbenv/versions/3.3.5/lib/ruby/gems/3.3.0/gems/rubocop-1.66.1/lib/rubocop/cop/commissioner.rb:87:in `investigate'
/Users/me/.rbenv/versions/3.3.5/lib/ruby/gems/3.3.0/gems/rubocop-1.66.1/lib/rubocop/cop/team.rb:168:in `investigate_partial'
/Users/me/.rbenv/versions/3.3.5/lib/ruby/gems/3.3.0/gems/rubocop-1.66.1/lib/rubocop/cop/team.rb:102:in `investigate'
/Users/me/.rbenv/versions/3.3.5/lib/ruby/gems/3.3.0/gems/rubocop-1.66.1/lib/rubocop/runner.rb:349:in `block in inspect_file'
/Users/me/.rbenv/versions/3.3.5/lib/ruby/gems/3.3.0/gems/rubocop-1.66.1/lib/rubocop/runner.rb:348:in `each'
/Users/me/.rbenv/versions/3.3.5/lib/ruby/gems/3.3.0/gems/rubocop-1.66.1/lib/rubocop/runner.rb:348:in `flat_map'
/Users/me/.rbenv/versions/3.3.5/lib/ruby/gems/3.3.0/gems/rubocop-1.66.1/lib/rubocop/runner.rb:348:in `inspect_file'
/Users/me/.rbenv/versions/3.3.5/lib/ruby/gems/3.3.0/gems/rubocop-1.66.1/lib/rubocop/runner.rb:291:in `block in do_inspection_loop'
/Users/me/.rbenv/versions/3.3.5/lib/ruby/gems/3.3.0/gems/rubocop-1.66.1/lib/rubocop/runner.rb:325:in `block in iterate_until_no_changes'
<internal:kernel>:187:in `loop'
/Users/me/.rbenv/versions/3.3.5/lib/ruby/gems/3.3.0/gems/rubocop-1.66.1/lib/rubocop/runner.rb:318:in `iterate_until_no_changes'
/Users/me/.rbenv/versions/3.3.5/lib/ruby/gems/3.3.0/gems/rubocop-1.66.1/lib/rubocop/runner.rb:287:in `do_inspection_loop'
/Users/me/.rbenv/versions/3.3.5/lib/ruby/gems/3.3.0/gems/rubocop-1.66.1/lib/rubocop/runner.rb:168:in `block in file_offenses'
/Users/me/.rbenv/versions/3.3.5/lib/ruby/gems/3.3.0/gems/rubocop-1.66.1/lib/rubocop/runner.rb:193:in `file_offense_cache'
/Users/me/.rbenv/versions/3.3.5/lib/ruby/gems/3.3.0/gems/rubocop-1.66.1/lib/rubocop/runner.rb:167:in `file_offenses'
/Users/me/.rbenv/versions/3.3.5/lib/ruby/gems/3.3.0/gems/rubocop-1.66.1/lib/rubocop/runner.rb:158:in `process_file'
/Users/me/.rbenv/versions/3.3.5/lib/ruby/gems/3.3.0/gems/rubocop-1.66.1/lib/rubocop/runner.rb:139:in `block in each_inspected_file'
/Users/me/.rbenv/versions/3.3.5/lib/ruby/gems/3.3.0/gems/rubocop-1.66.1/lib/rubocop/runner.rb:138:in `each'
/Users/me/.rbenv/versions/3.3.5/lib/ruby/gems/3.3.0/gems/rubocop-1.66.1/lib/rubocop/runner.rb:138:in `reduce'
/Users/me/.rbenv/versions/3.3.5/lib/ruby/gems/3.3.0/gems/rubocop-1.66.1/lib/rubocop/runner.rb:138:in `each_inspected_file'
/Users/me/.rbenv/versions/3.3.5/lib/ruby/gems/3.3.0/gems/rubocop-1.66.1/lib/rubocop/runner.rb:124:in `inspect_files'
/Users/me/.rbenv/versions/3.3.5/lib/ruby/gems/3.3.0/gems/rubocop-1.66.1/lib/rubocop/runner.rb:77:in `run'
/Users/me/.rbenv/versions/3.3.5/lib/ruby/gems/3.3.0/gems/rubocop-1.66.1/lib/rubocop/cli/command/execute_runner.rb:26:in `block in execute_runner'
/Users/me/.rbenv/versions/3.3.5/lib/ruby/gems/3.3.0/gems/rubocop-1.66.1/lib/rubocop/cli/command/execute_runner.rb:52:in `with_redirect'
/Users/me/.rbenv/versions/3.3.5/lib/ruby/gems/3.3.0/gems/rubocop-1.66.1/lib/rubocop/cli/command/execute_runner.rb:25:in `execute_runner'
/Users/me/.rbenv/versions/3.3.5/lib/ruby/gems/3.3.0/gems/rubocop-1.66.1/lib/rubocop/cli/command/execute_runner.rb:17:in `run'
/Users/me/.rbenv/versions/3.3.5/lib/ruby/gems/3.3.0/gems/rubocop-1.66.1/lib/rubocop/cli/command.rb:11:in `run'
/Users/me/.rbenv/versions/3.3.5/lib/ruby/gems/3.3.0/gems/rubocop-1.66.1/lib/rubocop/cli/environment.rb:18:in `run'
/Users/me/.rbenv/versions/3.3.5/lib/ruby/gems/3.3.0/gems/rubocop-1.66.1/lib/rubocop/cli.rb:122:in `run_command'
/Users/me/.rbenv/versions/3.3.5/lib/ruby/gems/3.3.0/gems/rubocop-1.66.1/lib/rubocop/cli.rb:129:in `execute_runners'
/Users/me/.rbenv/versions/3.3.5/lib/ruby/gems/3.3.0/gems/rubocop-1.66.1/lib/rubocop/cli.rb:51:in `block in run'
/Users/me/.rbenv/versions/3.3.5/lib/ruby/gems/3.3.0/gems/rubocop-1.66.1/lib/rubocop/cli.rb:81:in `profile_if_needed'
/Users/me/.rbenv/versions/3.3.5/lib/ruby/gems/3.3.0/gems/rubocop-1.66.1/lib/rubocop/cli.rb:43:in `run'
/Users/me/.rbenv/versions/3.3.5/lib/ruby/gems/3.3.0/gems/rubocop-1.66.1/exe/rubocop:19:in `<top (required)>'
/Users/me/.rbenv/versions/3.3.5/bin/rubocop:25:in `load'
/Users/me/.rbenv/versions/3.3.5/bin/rubocop:25:in `<top (required)>'
/Users/me/.rbenv/versions/3.3.5/lib/ruby/3.3.0/bundler/cli/exec.rb:58:in `load'
/Users/me/.rbenv/versions/3.3.5/lib/ruby/3.3.0/bundler/cli/exec.rb:58:in `kernel_load'
/Users/me/.rbenv/versions/3.3.5/lib/ruby/3.3.0/bundler/cli/exec.rb:23:in `run'
/Users/me/.rbenv/versions/3.3.5/lib/ruby/3.3.0/bundler/cli.rb:455:in `exec'
/Users/me/.rbenv/versions/3.3.5/lib/ruby/3.3.0/bundler/vendor/thor/lib/thor/command.rb:28:in `run'
/Users/me/.rbenv/versions/3.3.5/lib/ruby/3.3.0/bundler/vendor/thor/lib/thor/invocation.rb:127:in `invoke_command'
/Users/me/.rbenv/versions/3.3.5/lib/ruby/3.3.0/bundler/vendor/thor/lib/thor.rb:527:in `dispatch'
/Users/me/.rbenv/versions/3.3.5/lib/ruby/3.3.0/bundler/cli.rb:35:in `dispatch'
/Users/me/.rbenv/versions/3.3.5/lib/ruby/3.3.0/bundler/vendor/thor/lib/thor/base.rb:584:in `start'
/Users/me/.rbenv/versions/3.3.5/lib/ruby/3.3.0/bundler/cli.rb:29:in `start'
/Users/me/.rbenv/versions/3.3.5/lib/ruby/gems/3.3.0/gems/bundler-2.5.16/exe/bundle:28:in `block in <top (required)>'
/Users/me/.rbenv/versions/3.3.5/lib/ruby/3.3.0/bundler/friendly_errors.rb:117:in `with_friendly_errors'
/Users/me/.rbenv/versions/3.3.5/lib/ruby/gems/3.3.0/gems/bundler-2.5.16/exe/bundle:20:in `<top (required)>'
/Users/me/.rbenv/versions/3.3.5/bin/bundle:25:in `load'
/Users/me/.rbenv/versions/3.3.5/bin/bundle:25:in `<main>'
```

## Steps to reproduce the problem

1. File affected

```ruby
class Share < Base
  attribute :type, Types::String.default("share_modal".freeze)
  attribute :label, Types::String.default("Share".freeze)
  attribute :resource_type, Types::String
  attribute :resource_name, Types::String.optional.default(nil)
  attribute :ownership_url, Types::String
  attribute :author_url, Types::String
  attribute :update_ownerships_base_url, Types::String
  attribute :id_accessor, Types::String.default("id".freeze)
  attribute :name_accessor, Types::String.default("name".freeze)
  attribute :general_description, Types::String.optional.default(nil)
  attribute :sharing_options_description, Types::String.optional.default(nil)
  attribute :title, Types::String.optional.default(nil)
  attribute :placeholder, Types::String.optional.default(nil)

  def initialize(attributes)
    attributes[:resource_name] ||= attributes[:resource_type].downcase
    attributes[:sharing_options_description] ||= attributes[:general_description]

    super
  end

  def to_h
    super.merge(
          resource_type: resource_type,
          resource_name: resource_name,
          ownership_url: ownership_url,
          author_url: author_url,
          update_ownerships_base_url: update_ownerships_base_url,
          id_accessor: id_accessor,
          name_accessor: name_accessor,
          general_description: general_description,
          sharing_options_description: sharing_options_description,
          title: title,
          placeholder: placeholder
        )
  end
end
```

## RuboCop version

$ [bundle exec] rubocop -V
1.66.1 (using Parser 3.3.6.0, rubocop-ast 1.36.1, running on ruby 3.3.5) [arm64-darwin23]
```

**Comment by koic** [comment]
This issue is a dup with #13454. Thank you.
